client/db/src/lib.rs

```diff
@@ -961,6 +977,28 @@ impl<Block: BlockT> Backend<Block> {
 		Self::from_database(db as Arc<_>, canonicalization_delay, &config)
 	}
 
+	/// Return the type of the database
+	pub fn database_type(source: DatabaseSettingsSrc) -> ClientResult<DatabaseType> {
+		let config = DatabaseSettings{
+			state_cache_size: 0,
+			state_cache_child_ratio: None,
+			state_pruning: PruningMode::ArchiveAll,
+			source: source,
+			keep_blocks: KeepBlocks::All,
+			transaction_storage: TransactionStorageMode::BlockBody
+		};
+
+		if crate::utils::open_database::<Block>(&config, DatabaseType::Full).is_ok() {
+			return Ok(DatabaseType::Full);
+        }
+
+		if crate::utils::open_database::<Block>(&config, DatabaseType::Light).is_ok() {
+			return Ok(DatabaseType::Light);
```

Comment:
This method of detecting database type is a bit hacky. It relies on the fact that the databases have different number of columns. Also unlike rocksdb,  paritydb does not care if you don't open all of the columns. So it is important to try for full database first, before trying as light. Could you please add a comment to that extend?

Reply:
Skimming through code I just read that open_database also call https://github.com/paritytech/substrate/blob/b7409a2257c7c231b485ee827daa73d05d303047/client/db/src/utils.rs#L290 so should be fine?

Reply:
Right, there's already a method for it

Reply:
This was the best way I could find to differentiate them, will add a comment for full disclosure.

Reply:
If you move migration code to `sc-client-db` you can just use `utils::check_database_type` instead.

Reply:
I remember, that was the problem.